GOVERNANCE.md

```diff
@@ -0,0 +1,458 @@
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues in their assigned area, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the 'top' are the maintainers, whose job is to define and realize the project vision of thin-edge.io.
+Maintainers should strive for agreement based on the following factors:
+
+- The vision of the whole project
+- Feasibility, accounting for future technical and cultural shifts
+- The health of the project going forward
+
+Overall, the maintainers should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the maintainers are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try to make sure to explore possible options together before calling for time-consuming solutions like a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If an exclusive choice has to be made (where it is not possible to entertain two conflicting approaches), and no clear side is more advantageous to pick, a vote should be held with the majority opinion being the final decision.
+
+- Some decisions are local, e.g. a plugin does not impact others with its behaviour, however it still needs to be a 'good citizen' in the project (CI, format, etc..)
+    - These should be clearly scoped in their respective project part
+    - However, if needed a 'higher up' decision can be requested if no consensus is achieved
+
+
+
+## Project Structure
+
+The project has a hierarchy to solve these two problems:
+
+1. How can users make sure that their voice and interests are represented in the project?
+    - Each member party appoints one core team member
+    - They form the group of people that have to, and are responsible for, signing off on the technical implementation of the project
+2. How to make the lives of these team members easier?
+
+
+-----
+
+* Thin-edge is organized around sub-projects - to ease decision taking.
+    * The sub-projects are organized in a hierarchy
+        * This makes it easier to decouple different parts of the project
+        * This makes it easier to decide on technical questions
+    * Everyone has to agree on the core project as it is the foundations to all others.
```

Comment:
I think this is the most important point. We need to make sure not only the maintainers agree on the core project but also all the people who leverage it (e.g. full-time contributors) feel comfortable with core project changes, as this might imply additional work and significant changes/rework for them.

Reply:
We need to have a structured suggestion process

Reply:
Follow-up task: https://github.com/thin-edge/thin-edge.io/issues/1377